### PR TITLE
xtoken use callback to deal with native token

### PR DESCRIPTION
**Pull request by xiaoch05**

### CODE REVIEW

File: helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        _;
+    }
+
+    modifier onlyXTokenBackingAuthorized() {
+        require(address(xTokenBacking) == msg.sender || xTokenBacking.guard() == msg.sender, "invalid sender");
+        _;
+    }
+
+    constructor(address _wToken, address _xTokenBacking) {
+        wToken = _wToken;
```
Comment (xiaoch05) [inline helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol:42]: require(wToken != address(0))

File: helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import "../interfaces/IXTokenBacking.sol";
+import "../interfaces/IXTokenCallback.sol";
+import "../../interfaces/IWToken.sol";
+import "../../../utils/TokenTransferHelper.sol";
+
+contract WTokenConvertor is IXTokenCallback, IXTokenRollbackCallback, ERC165 {
+    address public immutable wToken;
+    IXTokenBacking public immutable xTokenBacking;
+
+    mapping(uint256=>address) senders;
```
Comment (xiaoch05) [inline helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol:16]: public

File: helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        IERC20(_wToken).approve(_xTokenBacking, type(uint256).max);
+    }
+
+    function xTokenCallback(
+        uint256 _transferId,
+        address _xToken,
+        uint256 _amount,
+        bytes calldata extData
+    ) onlyXTokenBackingAuthorized external {
+        address recipient = address(bytes20(extData));
```
Comment (xiaoch05) [inline helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol:53]: abi.decode

Comment (xiaoch05) [reply helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol:53]: docs for encoding of extData

**Review by hackfisher — APPROVED** [review]

File: helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        _;
+    }
+
+    modifier onlyXTokenBackingAuthorized() {
+        require(address(xTokenBacking) == msg.sender || xTokenBacking.guard() == msg.sender, "invalid sender");
+        _;
+    }
+
+    constructor(address _wToken, address _xTokenBacking) {
+        require(_wToken != address(0), "invalid wtoken address");
```
Comment (xiaoch05) [inline helix-contract/contracts/xtoken/v3/templates/WTokenConvertor.sol:42]:
```suggestion
        require(_wToken.code.length > 0, "invalid wtoken address");
```

**Pull request merged**